src/main/java/pages/ProductPage.java

```diff
@@ -1,30 +1,44 @@
 package pages;
 
+import lombok.Getter;
 import org.openqa.selenium.WebDriver;
 import org.openqa.selenium.WebElement;
 import org.openqa.selenium.support.FindBy;
 
-import lombok.Getter;
-
 @Getter
 public class ProductPage extends PageObject {
 
     //TODO: add URL for product details page
     private String pageUrl = "";
 
-    @FindBy(xpath = "//input[@id='add-to-cart-button-19']")
+    @FindBy( xpath = "//input[@id='add-to-cart-button-19']")
     private WebElement addToCartButton;
 
-    @FindBy(xpath = "//input[@id='add-to-wishlist-button-19']")
-    private WebElement addToWishButton;
+    @FindBy( css = " .add-to-wishlist input")
+    private WebElement addToWishlistButton;
 
-    @FindBy(xpath = "//span[@class='cart-label']")
+    @FindBy( xpath = "//span[@class='cart-label']")
     private WebElement ShoppingCartButton;
 
-    @FindBy(xpath = "//span[@class='wishlist-label']")
+    @FindBy( xpath = "//span[@class='wishlist-label']")
     private WebElement WishListButton;
 
-    public ProductPage(final WebDriver driver) {
+    @FindBy( xpath = "//a[contains(text(),'wishlist')]")
+    private WebElement WishListLink;
+
+    @FindBy(css = ".bar-notification.success")
+    private WebElement alertMessage;
+
+    @FindBy(css = "span.close")
+    private WebElement closeAlertMessage;
+
+    @FindBy( css = " .product-name > h1")
+    private WebElement productName;
+
+    @FindBy(css = " .header .header-upper .header-links-wrapper .header-links")
+    private WebElement headerForm;
```

Comment:
recheck webElements from this page
there are some elements which should be placed at abstractPage

Reply:
Done